src/main/java/data_access/UserDataAccessObject.java

```diff
@@ -19,70 +18,98 @@
 public class UserDataAccessObject implements ClubSignupUserDataAccessInterface, StudentSignupUserDataAccessInterface,
         ClubLoginDataAccessInterface, StudentLoginDataAccessInterface, ClubCreatePostUserDataAccessInterface {
 
-    private final ArrayList<User> userArrayList = new ArrayList<>();
+    private final ArrayList<Student> studentArrayList = new ArrayList<>();
+    private final ArrayList<Club> clubArrayList = new ArrayList<>();
 
     @Override
-    public boolean existsByName(String identifier) {
+    public boolean existsByNameClub(String identifier) {
         boolean found = false;
-        for (User user : userArrayList) {
-            if (user.getUsername().equals(identifier)) {
+        for (Club club : clubArrayList) {
+            if (club.getUsername().equals(identifier)) {
                 found = true;
+                break;
+            }
+        }
+        return found;
+    }
+
+    @Override
+    public boolean existsByNameStudent(String username) {
+        boolean found = false;
+        for (Student student : studentArrayList) {
+            if (student.getUsername().equals(username)) {
+                found = true;
+                break;
             }
         }
         return found;
     }
 
     @Override
```

Comment:
good job splitting into club and user cases, makes a lot more sense.

Reply:
Thanks